classes/core/SettingsBuilder.php

```diff
@@ -0,0 +1,341 @@
+<?php
+
+/**
+ * @file classes/core/SettingsBuilder.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SettingsBuilder
+ *
+ * @brief The class that extends Eloquent's builder to support settings tables for Models
+ */
+
+namespace PKP\core;
+
+use Closure;
+use Illuminate\Contracts\Database\Query\ConditionExpression;
+use Illuminate\Contracts\Database\Query\Expression;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Query\Builder as QueryBuilder;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Str;
+use stdClass;
+
+class SettingsBuilder extends Builder
+{
+    /**
+     * Get the hydrated models without eager loading.
+     *
+     * @param  array|string  $columns
+     *
+     * @return \Illuminate\Database\Eloquent\Model[]|static[]
+     */
+    public function getModels($columns = ['*'])
+    {
+        $rows = $this->getModelWithSettings($columns);
+        $returner = $this->model->hydrate(
+            $rows->all()
+        )->all();
+
+        return $returner;
+    }
+
+    /**
+     * Update records in the database, including settings
+     *
+     * @return int
+     */
+    public function update(array $values)
+    {
+        // Separate Model's primary values from settings
+        [$settingValues, $primaryValues] = collect($values)->partition(
+            fn (mixed $value, string $key) => in_array(Str::camel($key), $this->model->getSettings())
+        );
+
+        // Don't update settings if they aren't set
+        if ($settingValues->isEmpty()) {
+            return parent::update($primaryValues->toArray());
+        }
+
+        $newQuery = clone $this->query;
+
+        if ($primaryValues->isNotEmpty()) {
+            $count = parent::update($primaryValues->toArray());
+        }
+
+        // TODO Eloquent transforms attributes to snake case, find and override instead of transforming here
+        $settingValues = $settingValues->mapWithKeys(
+            fn (mixed $value, string $key) => [Str::camel($key) => $value]
```

Comment:
It seams there is additional conversion of attributes between camel case and snake case happening somewhere in the Eloquent core. Not critical, can be handled later